plugins/redfish/fu-redfish-client.c

```diff
@@ -100,11 +100,69 @@ fu_redfish_client_coldplug_inventory (FuRedfishClient *self,
 				      GError **error)
 {
 	JsonArray *members;
+	JsonObject *member_id;
+	const gchar *member_uri;
+	JsonNode *node_root;
+	JsonObject *member;
+
 	members = json_object_get_array_member (inventory, "Members");
 	for (guint i = 0; i < json_array_get_length (members); i++) {
-		JsonObject *member = json_array_get_object_element (members, i);
-		if (!fu_redfish_client_coldplug_member (self, member, error))
+		g_autoptr(JsonParser) parser = json_parser_new ();
+		g_autoptr(GBytes) blob = NULL;
+
+		member_id = json_array_get_object_element (members, i);
+		member_uri = json_object_get_string_member (member_id, "@odata.id");
+		if (member_uri == NULL) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_FOUND,
+					     "no @odata.id string");
+			g_object_unref (parser);
```

Comment:
you never need to unref the parser, the `g_autoptr(JsonParser)` does this for you when it goes out of scope.

Reply:
Oh, great. Nice to know that :D